### ISSUE TITLE
Option to add binaries to path

### ISSUE THREAD

**Issue by JasonGross** [avsm/setup-ocaml#70]
It would be nice if I could set some variable to `true` to have the opam environment added to path, so I don't need to `opam exec` everything.  (It might be the case that running `opam env >> $GITHUB_ENV` is enough?)

**Comment by JasonGross** [comment]
Actually, I think you need `opam env | sed "s/'//g; s/; export .*//g" >> $GITHUB_ENV`

**Comment by dra27** [comment]
It's _better_ to use `opam exec` (because it's explicit), but I agree that this would be a useful feature

**Comment by smorimoto** [comment]
I agree with @dra27, `opam exec` is better. We can get rid of this by passing `--enable-shell-hook` to `opam init`, but still the behavior is often unstable.

**Comment by dra27** [comment]
`--enable-shell-hook` won't work for the Windows shells (yet) - it only works for bash, etc.

**Comment by avsm** [comment]
I think @JasonGross posted the answer in his followup comment
```
opam env | sed "s/'//g; s/; export .*//g" >> $GITHUB_ENV
```

Modifying GITHUB_ENV is sufficient for future steps in the workflow to inherit the env, and it would simplify subsequent commands. It might be best to make this a non-default option in the workflow config though, in case it has unexpected consequences (such as an opam switch in a future step nullifying it).

**Comment by smorimoto** [comment]
Of course I can add something to the action that handles this well, but I also think it would be smarter to add a better fix to opam after this summer.

**Comment by smorimoto** [comment]
I just opened a PR to address this. Since v2 removed quite a lot of the differences between each platform, it was done in a fairly simple way.

**Comment by rr0gi** [comment]
having opam environment set up would allow to not duplicate build commands in yml, but invoke same wrapper as developers are using (e.g. make)

**Comment by smorimoto** [comment]
Since the exact version of opam is available on all platforms, `opam exec` will no longer be needed by default in the next release! (in other words, v3!)
See #794
